model/cloud_client.go

```diff
@@ -1,11 +1,25 @@
 package model
 
 import (
+	"os"
+
 	cloudModel "github.com/mattermost/mattermost-cloud/model"
 )
 
-// NewCloudClientWithOAuth creates a new cloud client with OAuth.
-func NewCloudClientWithOAuth(address, clientID, clientSecret, tokenEndpoint string) *cloudModel.Client {
+// NewCloudClient creates a new cloud client with OAuth.
+func NewCloudClient(address, clientID, clientSecret, tokenEndpoint, apiKey string) *cloudModel.Client {
 	var headers map[string]string
+
+	if os.Getenv("MATTERWICK_LOCAL_TESTING") == "true" {
+		return cloudModel.NewClient(address)
+	}
+
+	if clientID != "" && clientSecret != "" && tokenEndpoint != "" {
```

Comment:
Maybe I am wrong here, but don't we want `if clientID == "" && clientSecret == "" && tokenEndpoint == "" {`?

Reply:
Yep, good catch